simulator/func_sim/trap_types.h

```diff
@@ -8,18 +8,54 @@
 
 // MIPT-MIPS modules
 #include <infra/types.h>
+#include "riscv.opcode.gen.h"
 
-enum class Trap : uint8
-{
-    NO_TRAP,
-    HALT,
-    EXPLICIT_TRAP,
-    BREAKPOINT,
-    SYSCALL,
-    INTEGER_OVERFLOW,
-    UNSUPPORTED_SYSCALL,
-    UNALIGNED_ADDRESS,
-    UNKNOWN_INSTRUCTION
+#include <array>
+#include <string_view>
+
+
+class Trap {
+    public:
+        enum TrapType : uint8
+        {
+            //NOLINTNEXTLINE(cppcoreguidelines-macro-usage)
+            #define TRAP(name) name,
+            #include <func_sim/trap_types.def>
+            #undef TRAP
+            MAX_TRAP_TYPE
+        };
+
+        constexpr Trap(TrapType id) : value(id) { }
+
+        operator uint8() const { return value; }
+
+        void set_from_gdb_format(uint8 id);
+        uint8 to_gdb_format();
+
+        void set_from_riscv_format(uint8 id);
+        uint8 to_riscv_format();
```

Comment:
These methods have to be covered by some unit tests (not necessary to cover all possible traps, but just few of them to get sure the code is working at all).

Reply:
https://github.com/MIPT-ILab/mipt-mips/blob/12313df0067449e6d99cebfdd8a2a5d9a396a31b/simulator/func_sim/t/unit_test.cpp#L183-L196